week-3/task2/WithTime.js

```diff
@@ -0,0 +1,23 @@
+import EventEmitter from '../task1/EventEmitter.js';
+
+export default class WithTime extends EventEmitter {
+    async execute(asyncFunc, ...args) {
```

Comment:
```suggestion
   execute(asyncFunc, ...args) {
```
You should implement this without change the function to async. It is not a good idea to mix functions based on events with the async approach